tests/trainer/test_trainer_callback.py

```diff
@@ -243,3 +247,76 @@ def test_event_flow(self):
                 callbacks=[MyTestTrainerCallback, MyTestTrainerCallback],
             )
             assert str(MyTestTrainerCallback) in warn_mock.call_args[0][0]
+
```

Comment:
Can we add a test/tests which pass in more than one callback (stateful and non-stateful) to make sure behaviour is as expected, and one which passes in two stateful callbacks and loads the correct state in.